controller/backing_image_manager_controller.go

```diff
@@ -649,12 +643,20 @@ func (c *BackingImageManagerController) prepareBackingImageFiles(currentBIM *lon
 			}
 			continue
 		}
+
 		bids, err := c.ds.GetBackingImageDataSource(biName)
 		if err != nil {
 			log.WithError(err).Warn("Failed to get backing image data source before preparing files, will skip handling this backing image")
 			continue
 		}
 
+		currentInfo, exists := currentBIM.Status.BackingImageFileMap[biName]
+		requireFile := !exists || currentInfo.State == longhorn.BackingImageStateFailed
+		requireFile = requireFile || bids.Status.CurrentState == longhorn.BackingImageStateFailed
+		if !requireFile {
+			continue
+		}
```

Comment:
```suggestion
		currentInfo, exists := currentBIM.Status.BackingImageFileMap[biName]
		requireFile := !exists || currentInfo.State == longhorn.BackingImageStateFailed
		// Ensure the bids can be deleted instead of being stuck in the `BackingImageStateFailed` state. 
		// ref: https://github.com/longhorn/longhorn/issues/6086#issuecomment-1590662594
		requireFile = requireFile || bids.Status.CurrentState == longhorn.BackingImageStateFailed
		if !requireFile {
			continue
		}
```